Fix WooCommerce PDF Invoices & Packing Slips placeholders compatibility

Spreeuw opened the pull request:
WooCommerce PDF Invoices & Packing Slips Premium Templates compatibility
The `{{myparcel_delivery_date}}` placeholder got left out in the 4.0 upgrade (while it was present in 3.x). I wasn't sure whether to place it in `includes/compatibility/class-wcpdf-compatibility.php` or ` includes/frontend/class-wcmp-frontend.php` but since the former appears to be used specifically for the track & trace placeholders I opted for the latter. Might be good to bundle them all together in the compatibility class after all?

Several bugs in the other placeholder methods are also addressed.

Fixes #412

EdieLemoine commented on includes/frontend/class-wcmp-frontend.php:
```diff
@@ -84,6 +81,23 @@ public function wpo_wcpdf_delivery_options($replacement, WC_Order $order)
 
+    /**
+     * @param $replacement
+     * @param $order
```
Could you write the param type(s) here?

EdieLemoine commented on includes/frontend/class-wcmp-frontend.php:
```diff
@@ -84,6 +81,23 @@ public function wpo_wcpdf_delivery_options($replacement, WC_Order $order)
+    public function wpo_wcpdf_delivery_date($replacement, WC_Order $order)
+    {
+        $deliveryOptions = WCMYPA_Admin::getDeliveryOptionsFromOrder($order);
+        if ( $deliveryDate = $deliveryOptions->getDate() ) {
```
Could you remove spaces inside `if ()` and move the assignment out of it?

Spreeuw commented:
2 additional bugs were reported and I decided to include them in this pull request because they're closely related:

1. the {{myparcel_tracktrace}} placeholder no longer worked - this turned out to have been accidentally renamed from you end. I have restored this (for backwards compatibility) and added the new placeholder name as a synonym
2. the filters in `includes/compatibility/class-wcpdf-compatibility.php` were linked to the wrong class name (without the namespace), I have replaced this by `__CLASS__`.

wahidnory commented:
This getting merged?

phemels commented:
@RichardPerdaan When will this be fixed? In version 4.1.2 it's still an issue :(

RichardPerdaan commented:
Because there were a lot of merge conflicts, when I changed it to master, I used the changes and generated a new PR (#474)

phemels commented:
MyParcel 4.1.4 geinstalleerd.
Krijg nu deze error als ik een pakbon wil maken:

![image](https://user-images.githubusercontent.com/29940263/100384997-c8738500-3021-11eb-98cf-b5fba3bc8349.png)
